packages/number-field/src/NumberField.ts

```diff
@@ -175,20 +184,25 @@ export class NumberField extends TextfieldBase {
     private _trackingValue = '';
     private lastCommitedValue?: number;
 
-    private setValue(value: number = this.value): void {
-        this.value = value;
+    private setValue(newValue: number = this.value): void {
+        // Capture previous value for accurate IME change detection
```

Comment:
For IME input, the value conversion process would cause `this.value` and `this.lastCommitedValue` to be identical at the time of the comparison, even though a change had happened. So I added `previousValue` to hold the value of `this.lastCommitedValue` before the update to `this.value`, which fixed it! :)